azure-pipelines.yml

```diff
@@ -25,22 +25,24 @@ jobs:
           yarn checkchange
         displayName: check change
 
+  - job: BuildTest
+    workspace:
+      clean: all
+    steps:
+      - template: azure-pipelines.tools.yml
+
       - script: |
-          yarn build
-        displayName: build
+          yarn
+        displayName: yarn
 
-      - task: ArchiveFiles@2
-        inputs:
-          rootFolderOrFile: $(Build.Repository.LocalPath)
-          # Include root folder so archive can just be extracted into $(Pipeline.Workspace)
-          includeRootFolder: true
-          archiveType: 'tar'
-          tarCompression: 'gz'
-          archiveFile: $(Build.ArtifactStagingDirectory)/build.tar.gz
-          replaceExistingArchive: true
+      - template: .devops/templates/pr-target-branch.yml
 
-      - publish: $(Build.ArtifactStagingDirectory)
-        artifact: Build-PR-$(Build.BuildNumber)
+      - script: |
+          yarn lage build test lint --since $(target_branch) --verbose
+        displayName: build, test, lint
+        env:
+          BACKFILL_CACHE_PROVIDER: 'azure-blob'
+          BACKFILL_CACHE_PROVIDER_OPTIONS: '{"connectionString":"$(BACKFILL_CONNECTION_STRING)", "container":"$(BACKFILL_CONTAINER)"}'
```

Comment:
Is this going to be a security issue to have these exposed in a public pipeline? (meaning is there anything else someone could do if they have the connection string?)

Reply:
I actually created a separate storage account just for this purpose. The leak would be carefully crafted hashed cache to bring about a code injection. Although this is already possible as a change in yml. 

For ultimate security, we should be doing a pipeline feature to not allow non collaborators forks to run by default... I think...

Reply:
This one could be pretty easily leaked by changing the yml or a build script and making a PR. (By comparison, that's not such a concern for tokens used in the release pipeline because you'd have to get a bad change checked in.)

If a feature exists which limits variable availability to only collaborators' PRs, that would be great, but I kind of doubt that exists. :(